core/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -296,6 +293,11 @@ public String toString() {
         }
     }
 
+    public boolean supersedes(ClusterState other) {
```

Comment:
I added java docs and a test. I think this is more natural than a static method and doesn't prevent us from testing?